hellohilt/src/main/java/com/airbnb/mvrx/hellohilt/HelloHiltViewModel.kt

```diff
@@ -0,0 +1,42 @@
+package com.airbnb.mvrx.hellohilt
+
+import com.airbnb.mvrx.BaseMvRxViewModel
+import com.airbnb.mvrx.MavericksViewModelFactory
+import com.airbnb.mvrx.MvRxState
+import com.airbnb.mvrx.hellohilt.di.AssistedViewModelFactory
+import com.airbnb.mvrx.hellohilt.di.hiltMavericksViewModelFactory
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+
+data class HelloHiltState(
+    val viewModelScopedClassId1: Int? = null,
+    val viewModelScopedClassId2: Int? = null,
+    val notViewModelScopedClassId1: Int? = null,
+    val notViewModelScopedClassId2: Int? = null,
+) : MvRxState
+
+class HelloHiltViewModel @AssistedInject constructor(
+    @Assisted state: HelloHiltState,
+    private val repo1: HelloRepository,
+    private val repo2: HelloRepository,
+) : BaseMvRxViewModel<HelloHiltState>(state) {
+
+    init {
+        setState {
+            copy(
+                viewModelScopedClassId1 = repo1.viewModelScopedClass.id,
+                viewModelScopedClassId2 = repo2.viewModelScopedClass.id,
+                notViewModelScopedClassId1 = repo1.notViewModelScopedClass.id,
+                notViewModelScopedClassId2 = repo2.notViewModelScopedClass.id,
+            )
+        }
+    }
+
+    @AssistedFactory
+    interface Factory : AssistedViewModelFactory<HelloHiltViewModel, HelloHiltState> {
+        override fun create(state: HelloHiltState): HelloHiltViewModel
```

Comment:
instead of needing to override this could be make the parent interface generic, ie `AssistedViewModelFactory<*,*>`. We convert to that when we multi bind anyway, so does having it be typed here matter? just thinking of ways to simplify the setup

Reply:
I don't think so because the return type has to match the `@AssistedInject` constructor